weld/common/src/main/java/org/jboss/as/weld/util/Reflections.java

```diff
@@ -74,10 +74,9 @@ public static boolean isAccessible(String className, ClassLoader classLoader) {
     public static boolean containsAnnotation(Class<?> javaClass, Class<? extends Annotation> requiredAnnotation) {
         for (Class<?> clazz = javaClass; clazz != null && clazz != Object.class; clazz = clazz.getSuperclass()) {
             // class level annotations
-            if (clazz == javaClass || requiredAnnotation.isAnnotationPresent(Inherited.class)) {
-                if (containsAnnotations(clazz.getAnnotations(), requiredAnnotation)) {
+            if ((clazz == javaClass || requiredAnnotation.isAnnotationPresent(Inherited.class))
+                    && containsAnnotations(clazz.getAnnotations(), requiredAnnotation)) {
```

Comment:
Notice here you are modifying the order of executions. On the left side, `containsAnnotations` is executed always if `clazz == javaClass || requiredAnnotation.isAnnotationPresent(Inherited.class)` is true, on the right side `containsAnnotations` is not executed in such case.

We should try to avoid these code changes, I would say it is harmless here, but if the side effect can be avoided, much better.

Reply:
Ups, sorry, my fault, I did not see that the OR was enclosed with the parenthesis